### ISSUE TITLE
time_id is reported to be NULL while loading results into postgres

### ISSUE THREAD

**Issue by gowthamrao** [OHDSI/CohortDiagnostics#872]
Warning/error messages while loading data into postgres

```
Caused by: org.postgresql.util.PSQLException: ERROR: null value in column "time_id" violates not-null constraint
  Detail: Failing row contains (6624, null, 8527004, 371170, 0.7492, 0.4335, optum_ehr_v1990_20220403).
```

**Comment by azimov** [comment]
This has been fixed in the develop branch already.

In future, please report bugs with the steps to reproduce and version of the package you are using.

**Comment by gowthamrao** [comment]
The warnings are still seen in develop branch

> 
> sql.BatchUpdateException: Batch entry 0 INSERT INTO epi_958_cd.temporal_covariate_value (cohort_id,covariate_id,database_id,mean,sd,sum_value,time_id) VALUES(9044.0,3.7016169801E10,'iqvia_pharmetrics_plus',0.0012,0.0351,1333.0,1) was aborted: ERROR: current transaction is aborted, commands ignored until end of transaction block  Call getNextException to see other errors in the batch.
> 	at org.postgresql.jdbc.BatchResultHandler.handleError(BatchResultHandler.java:169)
> 	at org.postgresql.jdbc.PgStatement.internalExecuteBatch(PgStatement.java:854)
> 	at org.postgresql.jdbc.PgStatement.executeBatch(PgStatement.java:893)
> 	at org.postgresql.jdbc.PgPreparedStatement.executeBatch(PgPreparedStatement.java:1644)
> 	at org.ohdsi.databaseConnector.BatchedInsert.executeBatch(BatchedInsert.java:114)
> Caused by: org.postgresql.util.PSQLException: ERROR: current transaction is aborted, commands ignored until end of transaction block
> 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)
> 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)
> 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323)
> 	at org.postgresql.jdbc.PgStatement.internalExecuteBatch(PgStatement.java:851)
> 	... 3 more
> Caused by: org.postgresql.util.PSQLException: ERROR: null value in column "time_id" violates not-null constraint
>   Detail: Failing row contains (8179, null, 2014006, 17, 0.1619, 0.3684, iqvia_pharmetrics_plus).
> 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)
> 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)
> 	at org.postgresql.core.v3.QueryExecutorImpl.flushIfDeadlockRisk(QueryExecutorImpl.java:1404)
> 	at org.postgresql.core.v3.QueryExecutorImpl.sendQuery(QueryExecutorImpl.java:1429)
> 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:507)
> 	at org.postgresql.jdbc.PgStatement.internalExecuteBatch(PgStatement.java:870)
> 	... 3 more
> ERROR: current transaction is aborted, commands ignored until end of transaction block
> java.sql.BatchUpdateException: Batch entry 0 INSERT INTO epi_958_cd.temporal_covariate_value (cohort_id,covariate_id,database_id,mean,sd,sum_value,time_id) VALUES(9044.0,2.106471501E9,'iqvia_pharmetrics_plus',7.0E-4,0.0263,747.0,10) was aborted: ERROR: current transaction is aborted, commands ignored until end of transaction block  Call getNextException to see other errors in the batch.
> 	at org.postgresql.jdbc.BatchResultHandler.handleError(BatchResultHandler.java:169)
> 	at org.postgresql.jdbc.PgStatement.internalExecuteBatch(PgStatement.java:854)
> 	at org.postgresql.jdbc.PgStatement.executeBatch(PgStatement.java:893)
> 	at org.postgresql.jdbc.PgPreparedStatement.executeBatch(PgPreparedStatement.java:1644)
> 	at org.ohdsi.databaseConnector.BatchedInsert.executeBatch(BatchedInsert.java:114)
> Caused by: org.postgresql.util.PSQLException: ERROR: current transaction is aborted, commands ignored until end of transaction block
> 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)
> 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)
> 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323)
> 	at org.postgresql.jdbc.PgStatement.internalExecuteBatch(PgStatement.java:851)
> 	... 3 more
> Caused by: org.postgresql.util.PSQLException: ERROR: null value in column "time_id" violates not-null constraint
>   Detail: Failing row contains (8179, null, 2014006, 17, 0.1619, 0.3684, iqvia_pharmetrics_plus).
> 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)
> 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)
> 	at org.postgresql.core.v3.QueryExecutorImpl.flushIfDeadlockRisk(QueryExecutorImpl.java:1404)
> 	at org.postgresql.core.v3.QueryExecutorImpl.sendQuery(QueryExecutorImpl.java:1429)
> 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:507)
> 	at org.postgresql.jdbc.PgStatement.internalExecuteBatch(PgStatement.java:870)
> 	... 3 more
> ERROR: current transaction is aborted, commands ignored until end of transaction block

i am not sure how i can share a reproducible example here, but here is the environment snapshot

```

iqvia_pharmetrics_plus,TM_2022-08-18 08:23:15,packageDependencySnapShotJson,"{
	""package"" : [
		""R"",
		""grDevices"",
		""graphics"",
		""utils"",
		""stats"",
		""methods"",
		""glue"",
		""cli"",
		""rlang"",
		""ellipsis"",
		""fansi"",
		""lifecycle"",
		""magrittr"",
		""pkgconfig"",
		""utf8"",
		""vctrs"",
		""bit"",
		""cpp11"",
		""crayon"",
		""fastmap"",
		""hms"",
		""pillar"",
		""prettyunits"",
		""purrr"",
		""R6"",
		""backports"",
		""bit64"",
		""cachem"",
		""generics"",
		""progress"",
		""Rcpp"",
		""tibble"",
		""tidyselect"",
		""tools"",
		""tzdb"",
		""withr"",
		""assertthat"",
		""blob"",
		""checkmate"",
		""clipr"",
		""DBI"",
		""dplyr"",
		""memoise"",
		""plogr"",
		""rJava"",
		""stringi"",
		""triebeard"",
		""vroom"",
		""dbplyr"",
		""jsonlite"",
		""readr"",
		""RSQLite"",
		""snow"",
		""SqlRender"",
		""stringr"",
		""urltools"",
		""xml2"",
		""zip"",
		""Andromeda"",
		""DatabaseConnector"",
		""digest"",
		""lubridate"",
		""ParallelLogger"",
		""RJSONIO"",
		""clock"",
		""CohortGenerator"",
		""FeatureExtraction"",
		""tidyr"",
		""CohortDiagnostics""
	],
	""version"" : [
		""4.2.1"",
		""4.2.1"",
		""4.2.1"",
		""4.2.1"",
		""4.2.1"",
		""4.2.1"",
		""1.6.2"",
		""3.3.0"",
		""1.0.4"",
		""0.3.2"",
		""1.0.3"",
		""1.0.1"",
		""2.0.3"",
		""2.0.3"",
		""1.2.2"",
		""0.4.1"",
		""4.0.4"",
		""0.4.2"",
		""1.5.1"",
		""1.1.0"",
		""1.1.1"",
		""1.8.0"",
		""1.1.1"",
		""0.3.4"",
		""2.5.1"",
		""1.4.1"",
		""4.0.5"",
		""1.0.6"",
		""0.1.3"",
		""1.2.2"",
		""1.0.9"",
		""3.1.7"",
		""1.1.2"",
		""4.2.1"",
		""0.3.0"",
		""2.5.0"",
		""0.2.1"",
		""1.2.3"",
		""2.1.0"",
		""0.8.0"",
		""1.1.3"",
		""1.0.9"",
		""2.0.1"",
		""0.2.0"",
		""1.0-6"",
		""1.7.8"",
		""0.3.0"",
		""1.5.7"",
		""2.2.1"",
		""1.8.0"",
		""2.1.2"",
		""2.2.15"",
		""0.4-4"",
		""1.9.2"",
		""1.4.0"",
		""1.7.3"",
		""1.3.3"",
		""2.2.0"",
		""0.6.1"",
		""5.0.4"",
		""0.6.29"",
		""1.8.0"",
		""3.0.1"",
		""1.3-1.6"",
		""0.6.1"",
		""0.6.0"",
		""3.2.0"",
		""1.2.0"",
		""3.0.3""
	]
}"
iqvia_pharmetrics_plus,TM_2022-08-18 08:23:15,argumentsAtDiagnosticsInitiationJson,"{
	""runInclusionStatistics"" : true,
	""runIncludedSourceConcepts"" : true,
	""runOrphanConcepts"" : true,
	""runTimeSeries"" : false,
	""runVisitContext"" : true,
	""runBreakdownIndexEvents"" : true,
	""runIncidenceRate"" : true,
	""runTemporalCohortCharacterization"" : true,
	""minCellCount"" : 5,
	""incremental"" : false,
	""temporalCovariateSettings"" : {
		"""" : [
			""::"",
			""FeatureExtraction"",
			""createTemporalCovariateSettings""
		],
		""useDemographicsGender"" : true,
		""useDemographicsAge"" : true,
		""useDemographicsAgeGroup"" : true,
		""useDemographicsRace"" : true,
		""useDemographicsEthnicity"" : true,
		""useDemographicsIndexYear"" : true,
		""useDemographicsIndexMonth"" : true,
		""useDemographicsIndexYearMonth"" : true,
		""useDemographicsPriorObservationTime"" : true,
		""useDemographicsPostObservationTime"" : true,
		""useDemographicsTimeInCohort"" : true,
		""useConditionOccurrence"" : true,
		""useProcedureOccurrence"" : true,
		""useDrugEraStart"" : true,
		""useMeasurement"" : true,
		""useConditionEraStart"" : true,
		""useConditionEraOverlap"" : true,
		""useConditionEraGroupStart"" : false,
		""useConditionEraGroupOverlap"" : true,
		""useDrugExposure"" : false,
		""useDrugEraOverlap"" : false,
		""useDrugEraGroupStart"" : false,
		""useDrugEraGroupOverlap"" : true,
		""useObservation"" : true,
		""useDeviceExposure"" : true,
		""useCharlsonIndex"" : true,
		""useDcsi"" : true,
		""useChads2"" : true,
		""useChads2Vasc"" : true,
		""useHfrs"" : false,
		""temporalStartDays"" : [
			""c"",
			[
				""-"",
				9999
			],
			[
				""-"",
				365
			],
			[
				""-"",
				180
			],
			[
				""-"",
				30
			],
			[
				""-"",
				365
			],
			[
				""-"",
				30
			],
			0,
			1,
			31,
			[
				""-"",
				9999
			]
		],
		""temporalEndDays"" : [
			""c"",
			0,
			0,
			0,
			0,
			[
				""-"",
				31
			],
			[
				""-"",
				1
			],
			0,
			30,
			365,
			9999
		]
	}
}"
iqvia_pharmetrics_plus,TM_2022-08-18 08:23:15,rversion,R version 4.2.1 (2022-06-23 ucrt)
iqvia_pharmetrics_plus,TM_2022-08-18 08:23:15,currentPackage,CohortDiagnostics
iqvia_pharmetrics_plus,TM_2022-08-18 08:23:15,currentPackageVersion,3.0.3
```

**Comment by gowthamrao** [comment]
See version is 3.0.3 which is the unreleased version (in develop). So i ask that we re-open and look into this

**Comment by gowthamrao** [comment]
Interestingly - the same csv files when loaded to postgres throw this warning but do not show this warning when loading to sqlite

and the postgres does not show characterization results

**Comment by gowthamrao** [comment]
More interesting - why is timeId NA here

![image](https://user-images.githubusercontent.com/13936600/185532438-07813f5f-8824-476f-8a68-accd1815a72c.png)

and in csv file here

![image](https://user-images.githubusercontent.com/13936600/185532680-3f44b495-f4dd-460b-817c-e0285489529a.png)


this is temporalCovariateValueDist

it is required https://github.com/OHDSI/CohortDiagnostics/blob/develop/inst/settings/resultsDataModelSpecification.csv#L202

**Comment by gowthamrao** [comment]
This does work https://github.com/OHDSI/CohortDiagnostics/blob/e6b5a7633d1320ca42d80713ce775c187043525f/R/Private.R#L247

**Comment by gowthamrao** [comment]
i cant reproduce it anymore - closing again
